site/app/models/Config.php

```diff
@@ -292,6 +296,31 @@ public function loadMasterConfigs($config_path) {
         if (!empty($this->semester) && !empty($this->course)) {
             $this->course_path = FileUtils::joinPaths($this->submitty_path, "courses", $this->semester, $this->course);
         }
+
+        // Get additional links to display in the global footer.
+        $footer_links_json_file = FileUtils::joinPaths($this->config_path, "footer_links.json");
+        if (file_exists($footer_links_json_file)) {
+            $data = file_get_contents($footer_links_json_file);
+            if ($data !== false) {
+                $footer_links_json_data = json_decode($data, true);
+                // Validate that every footer link ($row) has required columns: 'url' and 'title'.
+                // $row can also have an 'icon' column, but it is optional.
+                foreach ($footer_links_json_data as $row) {
+                    switch (false) {
+                    case array_key_exists('url', $row):
+                    case array_key_exists('title', $row):
+                        //Validation fail.  Exclude $row.
+                        continue;
+                    default:
+                        //Validation OK.  Include $row.
+                        if (isset($row['icon']) && !Utils::startsWith($row['icon'], "fa-")) {
+                            $row['icon'] = "fa-" . $row['icon'];
+                        }
+                        $this->footer_links[] = $row;
+                    }
+                }
+            }
+        }
```

Comment:
I'm not sure this array is something we're going to use reference beyond just the one time you have it so config is probably not the right place for it. I think it might make more sense to just make `getFooterLinks()` be a function off of `app/libraries/Output.php` that just opens the file, reads from it, and returns the resultant array of footer links (and not bother storing it anywhere as we just use it the once. We can refactor for performance if we find we're getting these links more often).

Reply:
I put the footer links in the system config class because the issue discussion agreed this was a system-level configuration.

Going right to the Output class would circumvent the GlobalController and GlobalView.  GlobalController and GlobalView are responsible for the footer.  Therefore I've move the Footer Links "configuration" to the GlobalController's `footer()` class.